### DISCUSSION TITLE
Taiga UI 4

### DISCUSSION THREAD

**Discussion by waterplea** [taiga-family/taiga-ui#4792] [RFC]
# 4.0 readiness

- [x] CDK (ready except for legacy)
- [x] Core (ready except for legacy and new textfields)
- [x] Kit
- [x] Addon Charts
- [x] Addon Commerce
- [x] Addon Doc
- [x] Addon Mobile
- [x] Addon Preview
- [x] Addon Table
- [x] Addon Tablebars
- [x] Experimental
- [x] Layout
- [x] Legacy (this package will keep old components that are hard to migrate for smoother transition)

https://github.com/taiga-family/taiga-ui/issues/5808

**Comment by loxy** [comment]
Hi!

Do you have a rough time schedule. So, will it happen this year or next year?

Best regards,
Kersten

**Reply by waterplea** [reply]
Not in Q3 definitely. Maybe late Q4 or early next year.

**Reply by waterplea** [reply]
Sorry to disappoint you but we had a lot of new components requested for our projects — hence a new `@taiga-ui/experimental` with a lot of new stuff that was taking our time and attention. Current timeline is the end of Q1 2024.

**Reply by jason-orbital** [reply]
Is there any guidance on a timeline for this?

**Reply by waterplea** [reply]
You can expect first release candidate by Monday

**Comment by waterplea** [comment]
4.0 is out, onward to the next adventure! https://github.com/orgs/taiga-family/discussions/8353
